include/xrpl/protocol/SField.h

```diff
@@ -355,329 +355,29 @@ using SF_XCHAIN_BRIDGE = TypedField<STXChainBridge>;
 
 //------------------------------------------------------------------------------
 
+// Use macros for most SField construction to enforce naming conventions.
+#pragma push_macro("UNTYPED_SFIELD")
+#undef UNTYPED_SFIELD
+
+#define UNTYPED_SFIELD(name, stiSuffix, fieldValue, ...) \
+    extern SField const sf##name;
+
+#pragma push_macro("TYPED_SFIELD")
+#undef TYPED_SFIELD
+
+#define TYPED_SFIELD(name, stiSuffix, fieldValue, ...) \
+    extern SF_##stiSuffix const sf##name;
+
```

Comment:
For consistency, please follow the same style as in `Feature.h` (order of operations where `#push_macro` and `#undef` go interleaved, then all new `#define`, then `#undef` and `#pop_macro` go interleaved)